src/auth/externalAccountAuthorizedUserClient.ts

```diff
@@ -0,0 +1,335 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {AuthClient} from './authclient';
+import {Headers, RefreshOptions} from './oauth2client';
+import {
+  ClientAuthentication,
+  getErrorFromOAuthErrorResponse,
+  OAuthClientAuthHandler,
+  OAuthErrorResponse,
+} from './oauth2common';
+import {BodyResponseCallback, DefaultTransporter} from '../transporters';
+import {
+  GaxiosError,
+  GaxiosOptions,
+  GaxiosPromise,
+  GaxiosResponse,
+} from 'gaxios';
+import {Credentials} from './credentials';
+import * as stream from 'stream';
+import {EXPIRATION_TIME_OFFSET} from './baseexternalclient';
+
+/**
+ * External Account Authorized User Credentials JSON interface.
+ */
+export interface ExternalAccountAuthorizedUserClientOptions {
+  type: string;
```

Comment:
Based on usage, it seems it's `type` should always be `EXTERNAL_ACCOUNT_AUTHORIZED_USER_TYPE` (since anything else would be an error).